### DISCUSSION TITLE
How to create in-page references

### DISCUSSION THREAD

**Discussion by tinocolopez-a** [McShelby/hugo-theme-relearn#585] [Q&A] [answered]
Hello!

While using the Hugo-learn theme I was able to have in-page reference links by doing the following. I have this since I have a table of contents at the beginning of each page and each page is relatively long to scroll through, so having these links help. However, this same code doesn't work with the relearn theme. 

```
[1. Introduction 5](#_Toc131510730)

1.  <span id="_Toc131510730" class="anchor"></span>Introduction
```

Does anybody know how to enable this? Or what is the right way to code this in the new relearn theme? I went through the relearn documentation, but didn't find anything. Apologies if there is documentation and I missed it, if you could point me to it that'd be awesome!

Any help would be greatly appreciated!

**Comment by McShelby** [comment]
Your example works for me in the `exampleSite`.

Without seeing your code it's hard to say whats wrong.

I suppose you are either using a different Hugo version and/or other config settings in your `config.toml`

**Reply by tinocolopez-a** [reply]
[config.txt](https://github.com/McShelby/hugo-theme-relearn/files/11983284/config.txt)

This is my config file, I don't know if some of the parameters might be conflicting causing the links to not work.

**Comment by McShelby** [comment, marked as answer]
Without seeing your complete project nor the resulting output I would assume you haven't set Goldmarks `unsafe=true` setting.

**Reply by tinocolopez-a** [reply]
ok yes, I added the following and now the links are working. 

`[markup]
  [markup.goldmark]
    [markup.goldmark.renderer]
    unsafe = true`

However, for some reason the menu on the left disappears when I click on a page. The ones with the archetype "chapter" still show it, but not the other ones.

**Start of file for "Campaign Planning" file**

`+++
title = "Campaign Planning"
menuTitle = "Campaign Planning"
date = 2023-06-19 14:14:54
weight = 10
pre = ""
+++

**Table of Contents**`

![image](https://github.com/McShelby/hugo-theme-relearn/assets/123647243/eb524eac-fd36-490c-897b-9c573af59aec)

![image](https://github.com/McShelby/hugo-theme-relearn/assets/123647243/6c2bde4d-186a-4bfd-aaf2-7d5640341f96)

**Comment by McShelby** [comment]
Without a link to your repo I can't help you

**Reply by tinocolopez-a** [reply]
I figured out what was wrong with my files, thank you so much for the help! I really appreciate it!

**Reply by McShelby** [reply]
Would you mind to write about? This may help others. Just in case.

**Reply by tinocolopez-a** [reply]
Yes, I'll write and post about it over the weekend
